src/main/java/com/mabl/integration/jenkins/MablRestApiClientImpl.java

```diff
@@ -223,6 +259,23 @@ public String getAppBaseUrl() {
         return appBaseUrl;
     }
 
+    @Override
+    public void checkConnection() throws IOException {
+        final String url = restApiBaseUrl + HEALTH_ENDPOINT;
+        final HttpResponse response = httpClient.execute(buildGetRequest(url));
+        final int statusCode = response.getStatusLine().getStatusCode();
+        switch (statusCode) {
+            case 200:
+                break;
+            case 401:
+                throw new IOException("invalid API key provided or the proxy connection requires authentication");
```

Comment:
Minor: can we capitalize these?
```suggestion
                throw new IOException("Invalid API key provided or the proxy connection requires authentication");
```